perfkitbenchmarker/publisher.py

```diff
@@ -640,6 +650,81 @@ def _deDotKeys(self, res):
     return res
 
 
+class InfluxDBPublisher(SamplePublisher):
+
+  def __init__(self, influx_uri=None, influx_db_name=None):
+    # set to default above in flags unless changed
+    self.influx_uri = influx_uri
+    self.influx_db_name = influx_db_name
+
+  def PublishSamples(self, samples):
+    successful_http_request_codes = [200, 202, 204]
+    sample = list(samples)
+    number_of_samples = 0
+    for sample in samples:
+      number_of_samples += 1
+      timestamp = int((10 ** 9) * sample['timestamp'])
+      measurement = 'perfkitbenchmarker'
+      tag_set_metadata = \
+          ', '.join(self.FormatSampleForInfluxDB(sample['metadata']))
+      tag_set = ','.join((tag_set_metadata, self.
+                          FormatSampleForInfluxDB({k: sample[k]
+                                                   for k in ('test',
+                                                             'official',
+                                                             'owner',
+                                                             'run_uri',
+                                                             'sample_uri')})))
+      field_set = '%s_%s=%s' % (self.FormatSampleForInfluxDB({k: sample[k]
+                                for k in ('metric', 'unit', 'value')}))
+      samples_constructed_body = (' '.join((','.join((measurement,
```

Comment:
Can you post a real example in the comments of how the final request body looks like? It's hard to follow what is the format of this string.